include/veriblock/entities/context.hpp

```diff
@@ -11,19 +11,57 @@ namespace altintegration {
 struct BtcBlock;
 struct VbkBlock;
 struct VTB;
-struct AltPayloads;
 
 struct VbkContext {
-  std::vector<BtcBlock> btc;
+  std::vector<std::shared_ptr<BtcBlock>> btc;
 
   bool empty() const noexcept { return btc.empty(); }
 };
 
+struct PartialVTB {
+  std::vector<std::shared_ptr<BtcBlock>> btc;
+  std::vector<std::shared_ptr<VbkBlock>> context_vbk;
+  std::shared_ptr<VbkBlock> containing;
+  BtcEndorsement endorsement;
+
+  /**
+   * Return a containing VbkBlock
+   * @return containing block
+   */
+  VbkBlock getContainingBlock() const;
+
+  /**
+   * Return true if contains endorsement data
+   * @return true if contains endorsement data
+   */
+  bool containsEndorsements() const;
+
+  /**
+   * Return BtcEndorsement of the PartialVbk
+   * @return endorsement BtcEndorsement
+   */
+  BtcEndorsement getEndorsement() const;
+
+  /**
+   * Return BtcEndorsement id of the endorsement
+   * @return endorsement.id BtcEndorsement::id_t
+   */
+  typename BtcEndorsement::id_t getEndorsementId() const;
+
+  // generate without VBK context blocks, only BtcEndorsement VbkBlock
+  // containing and btc blocks
+  static PartialVTB generateFromVtb(const VTB& vtb);
```

Comment:
If PartialVTB does not contain VBK context blocks why do you have context_vbk in the struct?

Reply:
This context blocks fills independently in the other place, and that context can be different with the context blocks from the VTB, so in this place I don not copy them.